KarhooSDK/Api/SDKConfiguration/KarhooSDKConfiguration.swift

```diff
@@ -15,11 +15,18 @@ public protocol KarhooSDKConfiguration {
     func authenticationMethod() -> AuthenticationMethod
 
     func analyticsProvider() -> AnalyticsProvider
+
+    /// Closure to provide new auth credentials for KarhooSDK usage, when the current one expires and there is no refresh token available.
+    func requestNewAuthenticationCredentials(callback: @escaping () -> Void)
 }
 
 public extension KarhooSDKConfiguration {
 
     func analyticsProvider() -> AnalyticsProvider {
         return DefaultAnalyticsProvider()
     }
+
+    func requestNewAuthenticationCredentials(callback: @escaping () -> Void) {
```

Comment:
If we want to make this mandatory implementation then we need to remove this default implementation.

@mateid-kr are we making it mandatory? What do you think?

Reply:
Decided to make it mandatory for all implementations